### ISSUE TITLE
Error: .enforce_rtctrl: missing expected [ 

### ISSUE THREAD

**Issue by asharma1304** [CiscoDevNet/terraform-provider-aci#468]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.15.5
on darwin_amd64
+ provider registry.terraform.io/ciscodevnet/aci v0.7.0

### APIC version and APIC Platform

* V 4.2.6h and on-prem

### Affected Resource(s)

* data.aci_l3_outside

### Terraform Configuration Files

```hcl
Relevant portion of the configuration. If I want the 'terraform plan' to succeed without the above error, this is the module I comment out. 

data "aci_l3_outside" "SharedL3Out-West-Pod1_RO" {
  tenant_dn  = "${data.aci_tenant.common.id}"
  name       = "SharedL3Out-West-Pod1_RO"
}

I also comment out the line in this module where the above is referenced (last item)

resource "aci_bridge_domain" "HXVe-IB-MGMT_BD" {
    tenant_dn                   = "${aci_tenant.HXV-E-Foundation.id}"
    description                 = "Infra - InBand Mgmt BD for HX Edge Cluster"
    name                        = "HXVe-IB-MGMT_BD"
    arp_flood                   = "no"
    ep_move_detect_mode         = "garp"
    host_based_routing          = "no"
    unicast_route               = "yes" 
    relation_fv_rs_ctx          = aci_vrf.HXV-E-Foundation_VRF.id  
    relation_fv_rs_bd_to_out    = [data.aci_l3_outside.SharedL3Out-West-Pod1_RO.id]
}

### Debug Output

Seems like a pretty straightforward issue so pasting the relevant portion of the log here. I can still upload the trace log to GitHub Gist if it is needed. 
------
2021-06-04T10:22:18.373-0400 [TRACE] Executing graph transform *terraform.OrphanResourceInstanceCountTransformer
2021-06-04T10:22:18.373-0400 [TRACE] Completed graph transform *terraform.OrphanResourceInstanceCountTransformer (no changes)
2021-06-04T10:22:18.373-0400 [TRACE] Executing graph transform *terraform.AttachStateTransformer
2021-06-04T10:22:18.373-0400 [TRACE] Completed graph transform *terraform.AttachStateTransformer (no changes)
2021-06-04T10:22:18.373-0400 [TRACE] Executing graph transform *terraform.TargetsTransformer
2021-06-04T10:22:18.373-0400 [TRACE] Completed graph transform *terraform.TargetsTransformer (no changes)
2021-06-04T10:22:18.373-0400 [TRACE] Executing graph transform *terraform.ReferenceTransformer
2021-06-04T10:22:18.373-0400 [TRACE] readResourceInstanceState: reading state for data.aci_l3_outside.SharedL3Out-West-Pod1_RO
2021-06-04T10:22:18.373-0400 [DEBUG] ReferenceTransformer: "data.aci_contract.Allow-Shared-L3Out" references: []
2021-06-04T10:22:18.373-0400 [TRACE] Completed graph transform *terraform.ReferenceTransformer (no changes)
2021-06-04T10:22:18.373-0400 [TRACE] Executing graph transform *terraform.RootTransformer
2021-06-04T10:22:18.373-0400 [TRACE] Completed graph transform *terraform.RootTransformer (no changes)
2021-06-04T10:22:18.373-0400 [TRACE] vertex "data.aci_contract.Allow-Shared-L3Out": entering dynamic subgraph
2021-06-04T10:22:18.373-0400 [TRACE] vertex "data.aci_contract.Allow-Shared-L3Out": starting visit (*terraform.NodePlannableResourceInstance)
2021-06-04T10:22:18.373-0400 [TRACE] vertex "data.aci_l3_outside.SharedL3Out-West-Pod1_RO": visit complete
2021-06-04T10:22:18.373-0400 [TRACE] readResourceInstanceState: reading state for data.aci_contract.Allow-Shared-L3Out
2021-06-04T10:22:18.374-0400 [TRACE] vertex "data.aci_l3_outside.SharedL3Out-West-Pod1_RO": dynamic subgraph encountered errors: .enforce_rtctrl: missing expected [
2021-06-04T10:22:18.374-0400 [TRACE] vertex "data.aci_l3_outside.SharedL3Out-West-Pod1_RO": visit complete
2021-06-04T10:22:18.374-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to prevRunState for data.aci_contract.Allow-Shared-L3Out
2021-06-04T10:22:18.374-0400 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for data.aci_contract.Allow-Shared-L3Out
2021-06-04T10:22:18.374-0400 [TRACE] vertex "data.aci_l3_outside.SharedL3Out-West-Pod1_RO (expand)": dynamic subgraph encountered errors: .enforce_rtctrl: missing expected [
2021-06-04T10:22:18.374-0400 [TRACE] vertex "data.aci_l3_outside.SharedL3Out-West-Pod1_RO (expand)": visit complete
2021-06-04T10:22:18.374-0400 [TRACE] readDataSource: Re-validating config for data.aci_contract.Allow-Shared-L3Out
2021-06-04T10:22:18.374-0400 [TRACE] GRPCProvider: ValidateDataResourceConfig
2021-06-04T10:22:18.375-0400 [TRACE] readDataSource: data.aci_contract.Allow-Shared-L3Out configuration is complete, so reading from provider
2021-06-04T10:22:18.375-0400 [TRACE] GRPCProvider: ReadDataSource

### Expected Behavior

terraform plan should successfully execute as it did with previously, when I was running Terraform ACI Provider v0.6 


### Actual Behavior

After upgrade to Terraform ACI provider to v0.7, I get the error listed in the title when I execute terraform plan. I also upgraded Terraform core to v15.5 from v15.4 before filing this issue here as per the note under 'Version' - upgrading core did not help. 

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. terraform plan 

Worked with Terraform ACI provider v0.6
Fails with Terraform ACI provider v0.7
Upgrading Terraform core from v15.4 to 15.5 had no impact.

**Comment by lhercot** [comment]
This is related to the breaking change we made on aci_l3_outside. (Documented in the changelog here: https://github.com/CiscoDevNet/terraform-provider-aci/blob/master/CHANGELOG.md)
When you do a plan or apply. The first thing that TF will do is refresh everything which will include the aci_l3_outside.

Did you try directly removing the aci_l3_outside item with "terraform state rm 'aci_l3_outside.SharedL3Out-West-Pod1_RO'" ? You will then have to import it or re-apply.
Other people also added the [] to the aci_l3_outside 'enforce_rtctrl' attribute.

We are trying not to introduce breaking changes but those were bugs we had to fix has they were not letting user properly configured that property.

This seems to be a duplicate of #456

**Comment by asharma1304** [comment]
I somehow missed this in the changelog. I did do a terraform state rm on two other errors I was getting (related to spanning tree interface policy) as it was more obvious from the error message that it was a 'state' issue. 

However, I was able to workaround this by commenting out the module that was reading in the pre-existing l3out (and the reference to it). Terraform plan executed successfully after that. I then applied this change and when I went back to the original configuration by uncommenting the module ...and no errors this time.
